pylzy/lzy/utils/grpc.py

```diff
@@ -249,3 +251,41 @@ async def inner(*args: Any, **kwargs: Any) -> T:
         return inner
 
     return decorator
+
+
+def build_headers(token: str) -> List[ClientInterceptor]:
+    return add_headers_interceptor({
+        "authorization": f"Bearer {token}",
+        "x-client-version": f"pylzy={__version__}"
+    })
+
+
+def redefine_errors(f: Callable[..., Awaitable]) -> Callable[..., Awaitable]:
+
+    @functools.wraps(f)
+    async def wrapper(*args, **kwargs):
+        try:
+            return await f(*args, **kwargs)
+        except AioRpcError as e:
+            data = e.trailing_metadata().get_all("x-supported-client-versions")
+            if data and len(data) > 0:
+                try:
+                    supported_versions = json.loads(data[0])
+                except Exception as ex:
+                    _LOG.warning("Cannot parse supported versions from server metadata: %s", ex, exc_info=True)
```

Comment:
ex and exc_info in one log message is excessive.
And probably by the message "from server metadata: %s" you probably want to log data[0] instead of ex.